Android packaging can fail with VibeUE enabled (`Unknown Cook Failure` / UnrealEditor-Cmd exit timeout)

﻿## Summary

A user is reporting that packaging for Android fails when the VibeUE plugin is enabled. The current workaround is to disable VibeUE, restart the editor, package, then re-enable VibeUE and restart again.

The visible packaging error is:

```text
PackagingResults: Error: Unknown Cook Failure
AutomationTool exiting with ExitCode=25 (Error_UnknownCookFailure)
```

## Reported behavior

Reporter: `sevalto`

Project/setup details from the shared log and Discord thread:
- Unreal Engine `5.7.4`
- Windows 11 host
- Android ASTC packaging target
- Project: `GoldenKeyPrototype`
- VibeUE plugin enabled during packaging
- Repro reportedly happens with both:
  - VibeUE installed through Fab as a system-wide install
  - VibeUE added as a git-cloned project plugin

Original report from Discord:

> For some reason I have a packaging issue when VibeUE plugin is loaded. I have tried the following things to solve the issue, but have not found a way to fix this yet.
> Only workaround is to disable VibeUE - restart - package - enable Vibe UE - restart - continue working. Very annoying when I have to test builds on Quest - eats a lot of time.
> This happens both with the plugin installed through Fab as a system wide install as well as a git clone project plugin.
> I have tried to exclude the plugin in settings packaging, dedicate it to be an Editor only plugin in the uplugin file, exclude it in the Build file, but it always seems to fail because the cook cannot shut down properly and the error I am getting is
> PackagingResults: Error: Unknown Cook Failure
> So what I think is happening is the http process is preventing the cook to finish, but have not been able to find a solution for that yet.
> Any ideas how to work around this? Is anybody else experiencing this?

## Discord thread follow-up

- `buckley603` replied that VibeUE should be editor-only and asked for a larger log file to debug.
- `sevalto` uploaded `cookLog.txt`.
- `exetorius` reported they could package and cook successfully with VibeUE enabled on their side and asked whether the UE version was 5.7.
- `buckley603` later noted that `exetorius` was testing Windows packaging while `sevalto` was packaging for Android, so this may be platform-specific.
- `exetorius` also inspected the log and suggested the failure may be related to memory pressure rather than a direct plugin shutdown failure.

## Important observations from the attached log

The log contains evidence for both of these possibilities:

### 1. VibeUE does start during cook

```text
LogTemp: Display: VibeUE Module has started
LogMCPServer: Error: SECURITY WARNING: VibeUE MCP server is starting with NO API key set...
```

### 2. The cook completes its work and VibeUE appears to shut down cleanly

```text
LogTemp: Display: VibeUE OnPreExit - cleaning up Python services
LogTemp: Display: VibeUE OnPreExit - C++ cleanup complete
LogTemp: Display: VibeUE Module has shut down
```

### 3. The failure happens because UnrealEditor-Cmd does not exit in time after cook

```text
Waited for 68.80s for output of C:\Program Files\Epic Games\UE_5.7\Engine\Binaries\Win64\UnrealEditor-Cmd.exe, some output may be missing; we gave up. Now waiting 5 minutes for process to exit.
Took 104.88s to run UnrealEditor-Cmd.exe, ExitCode=1
Cook failed.
AutomationTool exiting with ExitCode=25 (Error_UnknownCookFailure)
```

### 4. There is also clear memory pressure during the same run

```text
Requested 1.5 GB memory per action, 1.27 GB available: limiting max parallel actions to 1
CookSettings.MemoryMinFreePhysical: Available physical memory 1953MiB is less than 2048MiB.
AssetCompile memory estimate is greater than available ... RequiredMemory = 2940.259277 MiB, MemoryLimit = 2366.367188 MiB
Peak Used physical 5157 MiB
```

## Why this issue is worth tracking

At minimum, this is a user-facing packaging problem that makes Quest/Android iteration much slower.

Even if memory pressure is the primary root cause, the report is still actionable because:
- VibeUE is active during cook despite being expected to behave as an editor-only plugin.
- The visible failure mode to the user is an `Unknown Cook Failure` when VibeUE is enabled.
- The process hang/timeout after cook completion makes the failure look like a plugin shutdown problem.
- Another tester reportedly succeeds on Windows, which suggests an Android-specific or environment-specific interaction that should be understood.

## Requested investigation

1. Verify whether VibeUE should be loaded at all during Android cook/package flows.
2. Check whether the MCP/Python services are contributing to delayed shutdown of `UnrealEditor-Cmd.exe` during cook.
3. Determine whether Android packaging with VibeUE enabled is expected to work on UE 5.7.x.
4. Confirm whether the observed `Unknown Cook Failure` is only secondary to low available physical memory, or whether VibeUE changes cook shutdown behavior.
5. If this is expected behavior under low-memory conditions, improve docs or detection so the failure mode is clearer to users.

## Attachments

- Full `cookLog.txt` is included in follow-up issue comments, split into chunks because it exceeds GitHub comment size limits.

﻿Full cook log from Discord attachment (cookLog.txt) part 3 of 3:

```text
9277 MiB, MemoryLimit = 2320.242188 MiB
UATHelper: Packaging (Android (ASTC)): LogAsyncCompilation: Display: BEWARE: AssetCompile memory estimate is greater than available, but we're running it [TextureDerivedData] anyway! RequiredMemory = 0.000 MiB + 2940.259277 MiB, MemoryLimit = 2320.679688 MiB
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorMaterials/PreviewShadowIndicator -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorBlueprintResources/StandardMacros -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorResources/S_Trigger -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorResources/S_BoxReflectionCapture -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooked packages 1676 Packages Remain 987 Total 2663
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorResources/S_Actor -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorMaterials/Camera/CineMat -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorLandscapeResources/SplineEditorMeshMat -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorResources/S_TriggerBox -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorResources/S_TriggerCapsule -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorResources/S_TriggerSphere -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorResources/S_Solver -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/VREditor/TransformGizmo/TranslucentTransformGizmoMaterial -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/VREditor/Sounds/UI/Drop_Item_From_ContentBrowser -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/VREditor/Sounds/UI/Gizmo_Handle_Clicked -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/VREditor/Sounds/UI/Object_PickUp -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/VREditor/Sounds/UI/Object_Snaps_To_Another_Actor -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/VREditor/Sounds/UI/Object_Snaps_To_Grid -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/VREditor/Sounds/UI/Scaling_Down -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/VREditor/Sounds/UI/Scaling_Up -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/VREditor/Sounds/UI/Selection_Changes -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/VREditor/LaserPointer/VR_LaserPower_01 -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorMaterials/ParticleSystems/PSysThumbnail_NoImage -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorMaterials/ParticleSystems/PSysThumbnail_OOD -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorMaterials/Dataflow/DataflowVertexMaterial -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorMaterials/Dataflow/DataflowTwoSidedVertexMaterial -> Rejected EngineEditorContent
UATHelper: Warning: Packaging (Android (ASTC)): LogBlueprint: Warning: [AssetLog] C:\Users\wille\Documents\Unreal Projects\GoldenKey-SeverinHP\Content\Blueprints\CBP_SandboxCharacter.uasset: [Compiler] Activate Persistent Global Camera Rig : Usage of 'Activate Persistent Global Camera Rig' has been deprecated. Use the similar functions on the GameplayCamera components or camera manager
UATHelper: Warning: Packaging (Android (ASTC)): LogBlueprint: Warning: [AssetLog] C:\Users\wille\Documents\Unreal Projects\GoldenKey-SeverinHP\Content\Blueprints\CBP_SandboxCharacter.uasset: [Compiler] Activate Persistent Base Camera Rig : Usage of 'Activate Persistent Base Camera Rig' has been deprecated. Use the similar functions on the GameplayCamera components or camera manager
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooked packages 1795 Packages Remain 868 Total 2663
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorMaterials/PhAT_NoCollisionMaterial -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorResources/S_Terrain -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorLandscapeResources/WhiteSquareTexture -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorLandscapeResources/DefaultAlphaTexture -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogTexture: Display: /Engine/EngineMaterials/BaseFlattenDiffuseMap_VT.BaseFlattenDiffuseMap_VT is marked for virtual streaming but virtual texture streaming is not available.
UATHelper: Packaging (Android (ASTC)): LogTexture: Display: /Engine/EngineMaterials/BaseFlattenDiffuseMap_VT.BaseFlattenDiffuseMap_VT is marked for virtual streaming but virtual texture streaming is not available.
UATHelper: Packaging (Android (ASTC)): LogTexture: Display: /Engine/EngineMaterials/BaseFlattenEmissiveMap_VT.BaseFlattenEmissiveMap_VT is marked for virtual streaming but virtual texture streaming is not available.
UATHelper: Packaging (Android (ASTC)): LogTexture: Display: /Engine/EngineMaterials/BaseFlattenEmissiveMap_VT.BaseFlattenEmissiveMap_VT is marked for virtual streaming but virtual texture streaming is not available.
UATHelper: Packaging (Android (ASTC)): LogTexture: Display: /Engine/EngineMaterials/BaseFlattenGrayscaleMap_VT.BaseFlattenGrayscaleMap_VT is marked for virtual streaming but virtual texture streaming is not available.
UATHelper: Packaging (Android (ASTC)): LogTexture: Display: /Engine/EngineMaterials/BaseFlattenGrayscaleMap_VT.BaseFlattenGrayscaleMap_VT is marked for virtual streaming but virtual texture streaming is not available.
UATHelper: Packaging (Android (ASTC)): LogTexture: Display: /Engine/EngineMaterials/BaseFlattenLinearColor_VT.BaseFlattenLinearColor_VT is marked for virtual streaming but virtual texture streaming is not available.
UATHelper: Packaging (Android (ASTC)): LogTexture: Display: /Engine/EngineMaterials/BaseFlattenLinearColor_VT.BaseFlattenLinearColor_VT is marked for virtual streaming but virtual texture streaming is not available.
UATHelper: Packaging (Android (ASTC)): LogTexture: Display: /Engine/EngineMaterials/BaseFlattenNormalMap_VT.BaseFlattenNormalMap_VT is marked for virtual streaming but virtual texture streaming is not available.
UATHelper: Packaging (Android (ASTC)): LogTexture: Display: /Engine/EngineMaterials/BaseFlattenNormalMap_VT.BaseFlattenNormalMap_VT is marked for virtual streaming but virtual texture streaming is not available.
UATHelper: Packaging (Android (ASTC)): LogTexture: Display: /Engine/EngineMaterials/Black_1x1_EXR_Texture_VT.Black_1x1_EXR_Texture_VT is marked for virtual streaming but virtual texture streaming is not available.
UATHelper: Packaging (Android (ASTC)): LogTexture: Display: /Engine/EngineMaterials/Black_1x1_EXR_Texture_VT.Black_1x1_EXR_Texture_VT is marked for virtual streaming but virtual texture streaming is not available.
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorResources/SequenceRecorder/Countdown -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorResources/SequenceRecorder/RecordingIndicator -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorMaterials/PhAT_UnselectedMaterial -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooked packages 1879 Packages Remain 784 Total 2663
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooked packages 2232 Packages Remain 431 Total 2663
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cook Diagnostics: OpenFileHandles=7618, VirtualMemory=4621MiB, VirtualMemoryAvailable=20557MiB
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooked packages 2583 Packages Remain 80 Total 2663
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNE_oidn2-3-0_rt_hdr_calb_cnrm_large__dyn__fp16_opset11_DenoiseAlpha -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNE_oidn2-3-0_rt_hdr_alb_nrm__dyn__fp16_opset11_ForwardAlpha -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNE_oidn2-3-0_rt_hdr_alb_nrm__dyn__fp16_opset11_DenoiseAlpha -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNE_oidn2-3-0_rt_hdr_alb_nrm_small__dyn__fp16_opset11_ForwardAlpha -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNE_oidn2-3-0_rt_hdr_alb_nrm_small__dyn__fp16_opset11_DenoiseAlpha -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNED_Oidn2-3_HighQuality_Alpha -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNED_Oidn2-3_HighQuality -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNED_Oidn2-3_Fast_Alpha -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNED_Oidn2-3_Fast -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNED_Oidn2-3_Balanced_Alpha -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNED_Oidn2-3_Balanced -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNEDT_Oidn2-3_Balanced_Alpha -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNEDT_Oidn2-3_Balanced -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNEDTOM_Output_Default -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNEDTOM_Output_Alpha -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNEDTIM_ColorAlbedoNormal_Default -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNEDTIM_ColorAlbedoNormal_Alpha -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNEDOM_Output_Default -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNEDOM_Output_Alpha -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNEDIM_HighQuality_Default -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNEDIM_HighQuality_Alpha -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNEDIM_ColorAlbedoNormal_Default -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNEDIM_ColorAlbedoNormal_Alpha -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /NNEDenoiser/NNE_oidn2-3-0_rt_hdr_calb_cnrm_large__dyn__fp16_opset11_ForwardAlpha -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Interchange/Pipelines/DefaultSceneAssetsPipeline -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Interchange/Pipelines/DefaultAudioPipeline -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Interchange/Pipelines/DefaultTexturePipeline -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Interchange/Pipelines/DefaultMaterialPipeline -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Interchange/Pipelines/DefaultGLTFPipeline -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Interchange/Pipelines/DefaultGLTFAssetsPipeline -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Interchange/Pipelines/DefaultMaterialXPipeline -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Interchange/Pipelines/DefaultAssetsPipeline -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Interchange/Pipelines/DefaultGLTFSceneAssetsPipeline -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Interchange/Pipelines/DefaultSceneLevelPipeline -> Rejected PlatformSpecificNeverCook
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooking /Engine/EditorResources/BSPVertex -> Rejected EngineEditorContent
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cooked packages 2663 Packages Remain 0 Total 2663
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cook Diagnostics: OpenFileHandles=10511, VirtualMemory=4865MiB, VirtualMemoryAvailable=20327MiB
UATHelper: Packaging (Android (ASTC)): LogAssetRegistryGenerator: Display: Saving asset registry v22.
UATHelper: Packaging (Android (ASTC)): LogAssetRegistryGenerator: Display: Generated development asset registry C:/Users/wille/Documents/Unreal Projects/GoldenKey-SeverinHP/Saved/Cooked/Android_ASTC/GoldenKeyPrototype/Metadata/DevelopmentAssetRegistry.bin num assets 2767, size is 2207.89kb, XxHash64[LE] = 0xAB87B338F61B3D7, waited on hash 0.00 seconds
UATHelper: Packaging (Android (ASTC)): LogAssetRegistryGenerator: Display: Generated asset registry num assets 2645, size is 965.33kb
UATHelper: Packaging (Android (ASTC)): LogShaderLibrary: Display:
UATHelper: Packaging (Android (ASTC)): LogShaderLibrary: Display: Shader Library 'GoldenKeyPrototype' (SF_VULKAN_ES31_ANDROID-VULKAN_ES3_1_ANDROID) Stats:
UATHelper: Packaging (Android (ASTC)): LogShaderLibrary: Display: =================
UATHelper: Packaging (Android (ASTC)): LogShaderLibrary: Display: Assets: 187, Unique Shadermaps: 166 (88.77%)
UATHelper: Packaging (Android (ASTC)): LogShaderLibrary: Display: Total Shaders: 2506, Unique Shaders: 1887 (75.30%)
UATHelper: Packaging (Android (ASTC)): LogShaderLibrary: Display: Total Shader Size: 13.52mb, Unique Shaders Size: 11.42mb (84.50%)
UATHelper: Packaging (Android (ASTC)): LogShaderLibrary: Display: === Extended info:
UATHelper: Packaging (Android (ASTC)): LogShaderLibrary: Display: Minimum number of shaders in shadermap: 1
UATHelper: Packaging (Android (ASTC)): LogShaderLibrary: Display: Median number of shaders in shadermap: 11
UATHelper: Packaging (Android (ASTC)): LogShaderLibrary: Display: Maximum number of shaders in shadermap: 181
UATHelper: Packaging (Android (ASTC)): LogShaderLibrary: Display: Number of shadermaps referencing top 10 most shared shaders:
UATHelper: Packaging (Android (ASTC)): LogShaderLibrary: Display:     45, 28, 21, 17, 17, 17, 17, 15, 15, 15
UATHelper: Packaging (Android (ASTC)): LogShaderLibrary: Display: Unique shaders itemization (sorted by compressed size):
UATHelper: Packaging (Android (ASTC)): LogShaderLibrary: Display: SF_Pixel: 1328 shaders (70.38%), compressed size: 9.11 MB (7.03 KB avg per shader), uncompressed size: 35.11 MB (27.07 KB avg per shader)
UATHelper: Packaging (Android (ASTC)): LogShaderLibrary: Display: SF_Vertex: 414 shaders (21.94%), compressed size: 1.80 MB (4.45 KB avg per shader), uncompressed size: 6.46 MB (15.98 KB avg per shader)
UATHelper: Packaging (Android (ASTC)): LogShaderLibrary: Display: SF_Compute: 145 shaders (7.68%), compressed size: 0.51 MB (3.60 KB avg per shader), uncompressed size: 1.86 MB (13.17 KB avg per shader)
UATHelper: Packaging (Android (ASTC)): LogShaderLibrary: Display: Textual dump saved to '../../../../../../Users/wille/Documents/Unreal Projects/GoldenKey-SeverinHP/Saved/Shaders/SF_VULKAN_ES31_ANDROID-VULKAN_ES3_1_ANDROID/ShaderDebug-GoldenKeyPrototype-SF_VULKAN_ES31_ANDROID-VULKAN_ES3_1_ANDROID/Dump.txt'
UATHelper: Packaging (Android (ASTC)): LogShaderLibrary: Display: =================
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Dependent plugin "SteamShared" referenced by "EOS Shared" wasn't found in enabled plugins list when creating cook metadata file... skipping
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Finalize package store(s)...
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Cook by the book total time in tick 33.454235s total time 36.054352
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Peak Used virtual 5334 MiB Peak Used physical 5157 MiB
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Packages Cooked: 2547, Packages Incrementally Skipped: 0, Packages Skipped by Platform: 116, Total Packages: 2663
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: Hierarchy Timer Information:
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:   Root: 0.000s (0)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     CleanSandbox: 0.068s (1)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     CollectFilesToCook: 0.004s (1)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       CookModificationDelegate: 0.000s (1)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     GenerateLongPackageName: 0.001s (1)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     TickMainCookLoop: 33.454s (3)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       PumpPollables: 0.002s (2)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       PumpRequests: 0.045s (84)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       LoadPackageForCooking: 16.802s (8541)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       ValidateSourcePackage: 0.000s (2663)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       SavingPackages: 15.083s (310)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:         PollPendingCookedPlatformDatas: 0.350s (7695)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:         PrepareSave: 3.321s (5233)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:         PrecachePlatformDataForNextPackage: 6.502s (2661)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:           PrepareSave: 6.500s (3170)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:         SaveCookedPackage: 4.707s (2663)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:           GEditorSavePackage: 4.395s (2561)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:             VerifyCanCookPackage: 0.014s (2547)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:         ClearAllCachedCookedPlatformData: 0.162s (2663)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       PostLoadPackageFixup: 0.000s (9)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       WaitForAsync: 1.008s (25186)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     CookByTheBookFinished: 0.960s (1)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       SavingCurrentIniSettings: 0.036s (1)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:         ProcessingAccessedStrings: 0.002s (1)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       ChunkGeneration: 0.010s (1)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       CalculateReferencedSet: 0.000s (1)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       FinishPopulateShaderLibrary: 0.001s (1)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       GeneratorPreSave: 0.000s (1)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       BuildChunkManifest: 0.036s (1)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:         FixupPackageDependenciesForChunks: 0.006s (1)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       SaveManifests: 0.000s (1)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       SaveAssetRegistry: 0.111s (1)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       WriteCookerOpenOrder: 0.085s (1)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       WriteCookMetadata: 0.001s (1)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       WriteReferencedSet: 0.003s (1)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:       FinalizePackageStore: 0.006s (1)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: Misc Cook Stats
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: ===============
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: Niagara
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     NiagaraScriptTranslationCount=0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     NiagaraCpuScriptCompileCount=0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     NiagaraCpuScriptDdcCacheHitCount=24
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     NiagaraCpuScriptDdcCacheMissCount=0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     NiagaraGpuScriptCompileCount=0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     NiagaraGpuScriptDdcCacheHitCount=0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     NiagaraGpuScriptDdcCacheMissCount=0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     NiagaraSystemWaitForCompilationCompleteTime=0.000147
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: ComputeKernelShader.Misc
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     ShadersCompiled=0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: NiagaraShader.Misc
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     ShadersCompiled=0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: ShaderCompiler
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     BlockingTimeSec=0.000068
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     AsyncCompileTimeSec=0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     GlobalBeginCompileShaderTimeSec=0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     GlobalBeginCompileShaderCalls=0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     ProcessAsyncResultsTimeSec=0.000173
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: GlobalShader.Misc
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     ShadersCompiled=0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: Material
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     FinishCacheShadersSec=0.249025
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: Package.Load
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     NumRequestedLoads=4173
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     NumPackagesLoaded=7331
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     NumInlineLoads=3158
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     LoadPackageTimeSec=16.989548
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: Package.Save
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     NumPackagesIncrementallySkipped=0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     NumPackagesSaved=2561
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     SavePackageTimeSec=4.123527
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     TagPackageExportsPresaveTimeSec=0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     TagPackageExportsTimeSec=0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     FullyLoadLoadersTimeSec=0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     ResetLoadersTimeSec=0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     TagPackageExportsGetObjectsWithOuter=0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     TagPackageExportsGetObjectsWithMarks=0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     SerializeImportsTimeSec=0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     SortExportsSeekfreeInnerTimeSec=0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     SerializeExportsTimeSec=0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     SerializeBulkDataTimeSec=0.025172
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     AsyncWriteTimeSec=0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     MBWritten=0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: CookOnTheFlyServer
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     PeakRequestQueueSize=2490
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     PeakLoadQueueSize=166
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     PeakSaveQueueSize=136
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: Package.DifferentPackagesSizeMBPerAsset
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: Package.NumberOfDifferencesInPackagesPerAsset
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: Package.PackageDifferencesSizeMBPerAsset
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: Package.DiffTotal
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     NumberOfDifferentPackages=0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     DifferentPackagesSizeMB=0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     NumberOfDifferencesInPackages=0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     PackageDifferencesSizeMB=0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: UnversionedProperties
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     SavedStructs=0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     SavedMB=0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     EquivalentTaggedMB=0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     CompressionRatio=-nan(ind)
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:     BitfieldWasteKB=0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: Cook Profile
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: ============
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  0.CookWallTimeSec=43.444220
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  0. 0.StartupWallTimeSec=6.769698
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  0. 1.StartCookByTheBookTimeSec=1.729179
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  0. 1. 0.BlockOnAssetRegistryTimeSec=0.028663
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  0. 1. 1.GameCookModificationDelegateTimeSec=0.000337
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  0. 2.TickCookOnTheSideTimeSec=18.180247
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  0. 2. 0.TickCookOnTheSideLoadPackagesTimeSec=16.804058
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  0. 2. 1.TickCookOnTheSideSaveCookedPackageTimeSec=4.707316
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  0. 2. 1. 0.TickCookOnTheSideResolveRedirectorsTimeSec=0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  0. 2. 2.TickCookOnTheSidePrepareSaveTimeSec=9.821853
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  0. 3.TickLoopGCTimeSec=0.520628
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  0. 4.TickLoopRecompileShaderRequestsTimeSec=0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  0. 5.TickLoopShaderProcessAsyncResultsTimeSec=0.000007
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  0. 6.TickLoopProcessDeferredCommandsTimeSec=0.000003
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  0. 7.TickLoopTickCommandletStatsTimeSec=0.000001
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  0. 8.TickLoopFlushRenderingCommandsTimeSec=0.002025
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  0. 9.ShaderFlushTimeSec=0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  0. 10.ValidationTimeSec=0.000767
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  1.CookParameters=
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  1. 0.TargetPlatforms=Android_ASTC
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  1. 1.CookProject=GoldenKeyPrototype
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  1. 2.CookCultures=
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  1. 3.IsCookAll=false
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  1. 4.IsCookOnTheFly=false
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  1. 5.IsIterativeCook=false
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  1. 6.IsUnversioned=true
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  1. 7.CookLabel=
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:  1. 8.IsFastCook=false
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: DDC Summary Stats
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: =================
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: BackEnd         =InstalledDerivedDataBackendGraph
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: HasLocalCache   =      true
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: HasSharedCache  =      true
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: HasCloudCache   =     false
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: HasZenCache     =      true
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: TotalGetHits    =      9923
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: TotalGetMisses  =         9
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: TotalGets       =      9932
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: TotalGetHitPct  =  0.999094
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: GetMissPct      =  0.000906
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: TotalPutHits    =         0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: TotalPutMisses  =         0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: TotalPuts       =         0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: TotalPutHitPct  =  0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: PutMissPct      =  0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: LocalGetHits    =      9510
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: LocalGetMisses  =        10
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: LocalGetTotal   =      9520
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: LocalGetHitPct  =  0.998950
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: SharedGetHits   =         1
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: SharedGetTotal  =        10
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: SharedGetHitPct =  0.100000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: ZenLocalGetHits =      9510
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: ZenLocalGetTotal=      9520
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: ZenLocalGetHitPct=  0.998950
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: ZenRemoteGetHits=         0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: ZenRemoteGetTotal=         0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: ZenRemoteGetHitPct=  0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: ZenRemoteReadSpeed=  0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: ZenRemoteWriteSpeed=  0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: ZenRemoteLatency=  0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: CloudGetHits    =         0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: CloudGetTotal   =         0
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: CloudGetHitPct  =  0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: LocalLatency    =  0.785690
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: LocalReadSpeed  =158.419897
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: LocalWriteSpeed =  0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: SharedLatency   =  0.683570
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: SharedReadSpeed =  6.249583
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: SharedWriteSpeed=  0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: CloudLatency    =  0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: CloudReadSpeed  =  0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: CloudWriteSpeed =  0.000000
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display:
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: DDC Resource Stats
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: =======================================================================================================
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: Asset Type                          Total Time (Sec)  GameThread Time (Sec)  Assets Built  MB Processed
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: ----------------------------------  ----------------  ---------------------  ------------  ------------
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: Texture                                         2.04                   0.00             0        698.02
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: BulkDataList                                    0.81                   0.00             0          0.69
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: AnimationSequence                               0.05                   0.00             0         70.01
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: DistanceField                                   0.04                   0.01             0          1.22
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: StaticMesh                                      0.04                   0.01             0         41.57
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: STREAMEDAUDIO                                   0.03                   0.00             0         17.98
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: CardRepresentation                              0.03                   0.00             0          0.03
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: NavCollision                                    0.02                   0.00             0          2.07
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: Audio                                           0.01                   0.01             0         17.79
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: MaterialShader                                  0.01                   0.24             0         18.72
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: GlobalShader                                    0.00                   0.01             0          7.93
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: PoseSearchDatabase                              0.00                   0.00             0         12.57
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: SkeletalMesh                                    0.00                   0.00             0         69.36
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: BodySetup                                       0.00                   0.00             0          8.95
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: NiagaraScript                                   0.00                   0.00             0          0.12
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: BulkDataPayloadId                               0.00                   0.00             0          0.00
UATHelper: Packaging (Android (ASTC)): LogCookStats: Display: CKERSM                                          0.00                   0.00             0          0.29
UATHelper: Packaging (Android (ASTC)): LogPackageBuildDependencyTracker: Display: Package Accesses (1661 referencing packages with a total of 83966 unique accesses)
UATHelper: Packaging (Android (ASTC)): LogConfigBuildDependencyTracker: Display: Config Accesses (905 referencing packages with a total of 5431 unique accesses). 1858 unique accesses that were not associated with a package.
UATHelper: Packaging (Android (ASTC)): LogCook: Display: Done!
UATHelper: Packaging (Android (ASTC)): LogSavePackage: Display: Took 0.003983s to verify the EDL loading graph.
UATHelper: Packaging (Android (ASTC)): LogInit: Display: Finished.
UATHelper: Packaging (Android (ASTC)): LogInit: Display:
UATHelper: Packaging (Android (ASTC)): 
UATHelper: Packaging (Android (ASTC)): Execution of commandlet took:  36.68 seconds
UATHelper: Packaging (Android (ASTC)): LogTemp: Display: VibeUE OnPreExit - cleaning up Python services
UATHelper: Packaging (Android (ASTC)): LogTemp: Display: VibeUE OnPreExit - C++ cleanup complete
UATHelper: Packaging (Android (ASTC)): LogTemp: Display: VibeUE Module has shut down
UATHelper: Packaging (Android (ASTC)): Waited for 68.80s for output of C:\Program Files\Epic Games\UE_5.7\Engine\Binaries\Win64\UnrealEditor-Cmd.exe, some output may be missing; we gave up. Now waiting 5 minutes for process to exit.
UATHelper: Packaging (Android (ASTC)): Took 104.88s to run UnrealEditor-Cmd.exe, ExitCode=1
UATHelper: Error: Packaging (Android (ASTC)): Cook failed.
(see C:\Users\wille\AppData\Roaming\Unreal Engine\AutomationTool\Logs\C+Program+Files+Epic+Games+UE_5.7\Log.txt for full exception trace)
UATHelper: Packaging (Android (ASTC)): AutomationTool executed for 0h 2m 33s
UATHelper: Packaging (Android (ASTC)): AutomationTool exiting with ExitCode=25 (Error_UnknownCookFailure)
UATHelper: Packaging (Android (ASTC)): BUILD FAILED
LogStreaming: Display: FlushAsyncLoading(376): 1 QueuedPackages, 0 AsyncPackages
LogConfig: Display: Audio Stream Cache "Max Cache Size KB" set to 0 by config: "../../../../../../Users/wille/Documents/Unreal Projects/GoldenKey-SeverinHP/Config/Engine.ini". Default value of 65536 KB will be used. You can update Project Settings here: Project Settings->Platforms->Windows->Audio->Cook Overrides->Stream Caching->Max Cache Size (KB)
PackagingResults: Error: Unknown Cook Failure
```

--- a/Source/VibeUE/Private/Module.cpp
+++ b/Source/VibeUE/Private/Module.cpp
@@ -115,6 +115,16 @@ void FModule::StartupModule()
 {
 	UE_LOG(LogTemp, Display, TEXT("VibeUE Module has started"));
 
+	// Skip all interactive services when running as a commandlet (e.g. -run=Cook).
+	// The MCP HTTP server and Python services keep threads alive that prevent
+	// UnrealEditor-Cmd.exe from exiting cleanly, causing UAT cook timeouts.
+	if (IsRunningCommandlet())
+	{
+		return;
+	}
+
+	bServicesInitialized = true;
+
 	// Clear screenshots directory from previous sessions to save disk space
 	FVibeUEPaths::ClearScreenshotsDir();
 
@@ -136,6 +146,12 @@ void FModule::StartupModule()
 
 void FModule::ShutdownModule()
 {
+	if (!bServicesInitialized)
+	{
+		UE_LOG(LogTemp, Display, TEXT("VibeUE Module has shut down"));
+		return;
+	}
+
 	// Unregister PreExit callback
 	FCoreDelegates::OnPreExit.RemoveAll(this);
 

--- a/Source/VibeUE/Public/Module.h
+++ b/Source/VibeUE/Public/Module.h
@@ -24,4 +24,9 @@ public:
 	{
 		return FModuleManager::Get().IsModuleLoaded("VibeUE");
 	}
+
+private:
+	// False when running as a commandlet (cook, etc.) — services are skipped to
+	// avoid keeping UnrealEditor-Cmd.exe alive after the commandlet finishes.
+	bool bServicesInitialized = false;
 }; 

